### ISSUE TITLE
Resolving LoadLibrary Error for Cross-Platform Compatibility

### ISSUE THREAD

**Issue by BinomeDeNewton** [secdev/scapy#4276]
### Brief description

When I ran a Scapy-based application on a Mac. The core of the problem was that Scapy attempted to load libc.so.6, a library specific to Linux systems, whereas macOS uses libc.dylib. This discrepancy led to a failure in executing Scapy-related functions, as the system could not find the required library, resulting in an OSError.

### Scapy version

2.5.0

### Python version

3.12

### Operating system

Apple M3 Pro

### How to reproduce

To ensure that the code runs seamlessly across different operating systems, a conditional import based on the operating system was implemented. I utilize the platform module to dynamically determine the operating system at runtime and then load the appropriate C library. For Linux, it continues to load libc.so.6, and for macOS, it switches to loading libc.dylib. 

```
import platform
ctypes.cdll.LoadLibrary("libc.{}".format("so.6" if platform.uname()[0] != "Darwin" else "dylib"))
libc = ctypes.CDLL("libc.{}".format("so.6" if platform.uname()[0] != "Darwin" else "dylib"))
```

### Actual result

```
OSError: dlopen(libc.so.6, 0x0006): tried: 'libc.so.6' (no such file), '/System/.../OSlibc.so.6' (no such file), '/usr/lib/libc.so.6' (no such file, not in dyld cache), 'libc.so.6' (no such file), '/usr/lib/libc.so.6' (no such file, not in dyld cache)
```

lines 2578-2579 : /scapy/layers/bluetooth.py

```
        # Properly close socket so we can free the device
        ctypes.cdll.LoadLibrary("libc.so.6")
        libc = ctypes.CDLL("libc.so.6")
```

lines 2536-2537 : /scapy/layers/bluetooth.py

```
        ctypes.cdll.LoadLibrary("libc.so.6")
        libc = ctypes.CDLL("libc.so.6")
```

### Expected result

After implementing the cross-platform solution for loading the correct version of the C standard library (`libc`) based on the operating system, the Python application should:

- Correctly load `libc.so.6` on Linux systems without any errors.
- On macOS, the application should seamlessly load `libc.dylib` instead of `libc.so.6`, avoiding the `OSError` related to the inability to find `libc.so.6`.
- Ensure that all functionality relying on the C standard library works as intended across different platforms, particularly between Linux and macOS.
- Eliminate any runtime errors related to the loading of system libraries, ensuring a smooth execution of the application's features that depend on `libc`.

**Comment by guedou** [comment]
Thanks for the report.

Can you test your issue against master?

Also, which version of macOS are you using?

**Comment by BinomeDeNewton** [comment]
I entered these commands to install scapy from Github instead of using pip:

```
git clone https://github.com/secdev/scapy
cd scapy
./run_scapy
bt = BluetoothHCISocket(0)
```

I encountered this error:
```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/.../Developer/scapy/scapy/layers/bluetooth.py", line 2506, in __init__
    s = socket.socket(socket.AF_BLUETOOTH, socket.SOCK_RAW, socket.BTPROTO_HCI)  # noqa: E501
                      ^^^^^^^^^^^^^^^^^^^
AttributeError: module 'socket' has no attribute 'AF_BLUETOOTH'
```

Indeed, macOS does not support `AF_BLUETOOTH` in Python's `socket` module. To work with Bluetooth in programming on macOS, you will need to use macOS-specific libraries or command-line tools like `bluetoothctl` (on Linux) or macOS-specific low-level programming interfaces. A good solution would be to use the `PyBluez` library. 

So, I am applying some modifications in the code on my Mac (version : 14.2.1 (23C71)).

**Comment by guedou** [comment]
Could you show us your modifications? I don't understand if you modified Scapy to stop the exception from happening or if you added Bluetooth support on macOS.

**Comment by BinomeDeNewton** [comment]
I found a way to address the issue with the libc library that Scapy encountered when running on a Mac. The core issue was Scapy's attempt to load `libc.so.6`, which is specific to Linux systems, while macOS uses `libc.dylib`. 

I implemented a conditional import based on the operating system. This solution utilizes the `platform` module to dynamically determine the operating system at runtime and then load the appropriate C library. For Linux, it continues to load `libc.so.6`, and for macOS, it switches to loading `libc.dylib`.

```python
import platform
import ctypes

ctypes.cdll.LoadLibrary("libc.{}".format("so.6" if platform.system() != "Darwin" else "dylib"))
libc = ctypes.CDLL("libc.{}".format("so.6" if platform.system() != "Darwin" else "dylib"))
```

As for the `socket.AF_BLUETOOTH` issue on macOS, I'm currently exploring solutions and considering alternatives like PyBluez.
